src/components/CodeInput/CodeInput.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {ReactNode, useCallback, useRef, useState, memo} from 'react'
+import {TextInput, Text, StyleProp, ViewStyle, KeyboardTypeOptions, TextStyle} from 'react-native'
+import {metrics} from '../../helpers/metrics'
+import styled from 'styled-components/native'
+import Cursor from './Cursor'
+import type {ITheme} from 'src/theme'
+
+interface CodeInputProps {
+  /** define style for cell */
+  cellStyle?: StyleProp<ViewStyle>
+
+  /** define style for cell when cell is focused */
+  focusCellStyle?: StyleProp<ViewStyle>
+
+  /** define style for text in the cell */
+  textStyle?: StyleProp<TextStyle>
+
+  /** define secure view style when using secureTextEntry mode */
+  secureViewStyle?: StyleProp<ViewStyle>
+
+  /** define style for text in the cell when cell is focused */
+  focusTextStyle?: StyleProp<TextStyle>
+
+  /** cell count */
+  length?: number
+  /** callback when complete  */
+
+  onSubmit?: (val: string) => void
+
+  /** render custom view for cursor/indicator */
+  customCursor?: () => ReactNode
+
+  /** render custom view for cursor/indicator */
+  secureTextEntry?: boolean
+
+  /** keyboard type */
+  keyboardType?: KeyboardTypeOptions
+}
+
+type CellStyle = {
+  theme: ITheme
+}
+
+type SecureViewStyle = {
+  theme: ITheme
+}
+
+const DEFAULT_LENGTH = 6
+
+const CodeInput = ({
+  cellStyle,
+  focusCellStyle,
+  textStyle,
+  focusTextStyle,
+  secureViewStyle,
+  length = DEFAULT_LENGTH,
+  onSubmit,
+  customCursor,
+  secureTextEntry,
+  keyboardType = 'number-pad',
+}: CodeInputProps) => {
+  const textInputRef = useRef<TextInput>()
+  const [code, setCode] = useState<string>('')
+
+  const handleOnChangeText = useCallback(
+    (val: string) => {
+      setCode(val)
+      if (val.length === length) {
+        onSubmit?.(val)
+        textInputRef.current?.blur()
+      }
+    },
+    [length, onSubmit],
+  )
+
+  const handleCellPress = useCallback(
+    (index: number) => {
+      if (index < code.length) {
+        setCode(code.slice(0, index))
+      }
+      textInputRef.current?.focus()
+    },
+    [code],
+  )
+
+  return (
+    <CodeInputContainer>
+      <StyledTextInput
+        testID="input"
+        ref={textInputRef}
+        value={code}
+        textContentType="oneTimeCode"
+        keyboardType={keyboardType}
+        onChangeText={handleOnChangeText}
+        maxLength={length}
+      />
+      <CellContainer>
+        {new Array(length).fill(0).map((_, index: number) => {
```

Comment:
how about `for (let i = 0; i < lenght; i++)`

Reply:
Can use for loop for this?

Reply:
Yes we can.

Reply:
Not sure what you mean. Could you give an example?

Reply:
my suggestion is create a function to render it to avoid re-render array too many times
`const renderInputs = useCallback(() => {
    const views = []
    for (let i = 0; i < lenght; i++) {
       views.push(<Input/>)
    }
    return views
}, [])`

Reply:
updated